### ISSUE TITLE
SparseAdam does not support with share_decoder_embeddings option

### ISSUE THREAD

**Issue by wonkeelee** [OpenNMT/OpenNMT-py#1015]
Hi, 

As far as I know is that to train transformer MT, set sparseadam to optimizer instead of adam through[ # 637](https://github.com/OpenNMT/OpenNMT-py/issues/637)
Therefore, When I tried to train transformer MT with `-share_decoder_embeddings` option, error occurred with message "SparseAdam does not support dense gradients, please consider Adam instead" 

Is there any problem with changing the optimizer to adam according to the error message?

Thanks!

**Comment by vince62s** [comment]
Did you use multi gpu or single gpu ?

in any event multi gpu does not support sparse adam so use adam instead.

but we may need to fix sparse adam single gpu with share_decoder_emneddings
we may have missed somehting.

**Comment by wonkeelee** [comment]
@vince62s  Thanks for your quick response. I worked with single GPU thanks!

**Comment by vince62s** [comment]
Can you please post your command line and error so that I can have a look ? (I think we may drop sparse adam in the future unless pytorch supports sparse for distributed functions)
